instrumentation_sql_test.go

```diff
@@ -349,3 +405,50 @@ type sqlRows2 struct{}
 func (sqlRows2) Columns() []string              { return []string{"col1", "col2"} }
 func (sqlRows2) Close() error                   { return nil }
 func (sqlRows2) Next(dest []driver.Value) error { return io.EOF }
+
+// Driver use case: driver does not implement NamedValueChecker,arg type checking is internal.
+// The idea is to mock pq: https://github.com/lib/pq/blob/8446d16b8935fdf2b5c0fe333538ac395e3e1e4b/encode.go#L31
+
+type sqlDriver3 struct{ Error error }
```

Comment:
Yet another mocked driver for a specific case, making the test file more verbose.
Maybe we can try to improve this part. 
Unfortunately, reusing a previous mocked driver with the embedding feature is not so simple, as there are more than one struct involved (driver, conn, statement, result, rows...), and usually each struct will have its own particularity due to the test being made.